docs/release-management.md

```diff
@@ -0,0 +1,86 @@
+# Proposal for release management in the submariner-io project
+
+## Introduction
+
+Releasing submariner contains numerous manual [steps](https://submariner-io.github.io/contributing/release-process/)
+that need to be executed carefully in order accross multiple projects,
+waiting for CI executions to properly happen.
+
+All this process could be automated as described in the following sections.
+
+## What's a release
+
+   A submariner release is the sum of all components, tagged on a specific set of versions,
+   that works, and it's tested together.
+   
+   The components include:
+   
+   * subctl
+   * submariner-operator
+   * submariner core (submariner engine, route agent, globalnet ipam)
+   * lighthouse (lighthouse agent, lighthouse coredns)
+   
+  
+## A release repository
+
+   A release repository (submariner-io/releases) could be used to track and automate
+   all the release process.
+   
+   Yaml files could be used inside this repository to control the definition of a release, 
+   
+   ```
+      releases/0.5-<name>/0.5.0-<name>.yaml
+      releases/0.5-<name>/0.5.1-bugfix for....yaml
+      ...
+   ```
+
+## A release yaml
+
+   A release yaml could have the following format:
+   
+   ```yaml
+    ---
+    version: 0.5.0
+    name: the mighty 0.5 release!
+    release-notes: 
+      - bugfixes: >
+        * we fixed this bug
+      - features: >
+        * this feature was added
+        * and this other one
+      - upgrades: >  # optional
+        Upgrade notes important to anybody upgrading from a previous version
+      - deprecations: > # optional
+        Features or settings that have been deprecated and need to be updated by the administrator.
+
+    components:
+      - submariner: <commit-id> 
+      - lighthouse: <commit-id>
+      - submariner-operator: <commit-id>
+      - submariner-charts: <commit-id>
+      - shipyard: <commit-id>
+
+   ```
+
+## verification of the release yaml
+
+The verification step would:
+* Verify the format of the yaml file
+* Verify that passed CI runs exist for the commit-ids proposed for the components
+* Run E2E for the combination of commit-ids with helm & operator (with lighthouse enabled)
+
+
+## merge of the release yaml
+
+* A script will run in Github actions, executing step by step the tagging, and watching
+of the multiple components.
+
+It will retry builds if necessary.
+
+* For the operator repository, a new commit will be automatically pushed with the versions.go
```

Comment:
But then you're not using the commit-id you originally had, you'll have another commit id since it's a new patch..

Reply:
Yeah, that is fine, we would know about it internally, and release on the new.

Reply:
The problem would be if something merges in the meanwhile... but hopefully we can control ourselves not to merge stuff.. and we can add a verification that our versions patch is being applied on top of the requested commit id

Reply:
But it's true, the release.yaml would not match with that...commit id, with it's itching, not sure if we can fix that somehow. May be inject those versions as env variables and have some defaults? The problem I see with that is that then.. if somebody tries to develop with subctl they need to use the right env vars... it would be tricky.

We could have subctl may be download the manifest?, or just assume that the versions are all equal to his own version by default?

Reply:
I think a more sensible approach would be to bump the versions POST the release, so that the master always had the next version (as is done in java maven builds, for example).

This way the hashes dont change and overall the flow is a bit simpler I guess..

Reply:
Hmm, that'd make sense, I like it, I will update the document.

Reply:
Cool, in fact the automated release process can bump these version and propose that as a PR to the necessary places :)

Reply:
I'll update and let's review